GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+    // check if proposal is correctly formatted. Includes fee payment.    
+    
+    if !correctlyFormatted(txClaimDeposit) then  
+      throw
+      
+    else 
+      if !exists(txClaimDeposit.ProposalID) then
+        // There is no proposal for this proposalID
+        
+        throw
+        
+      else 
+        retrieve depositorsList from txClaimDeposit.ProposalID
+        
+        
+        if sender is not in depositorsList then
+          throw
+          
+        else
+          retrieve deposit from sender in depositorsList 
+          
+          if deposit <= 0
+            // deposit has already been claimed
+            
+            throw
+            
+          else
+            retrieve proposal from txClaimDeposit.ProposalID
+            
+            if proposal.VotingStartBlock <= 0
+            // Vote never started
+            
+              if (CurrentBlock <= proposal.SubmitBlock + ActiveProcedure.MaxDepositPeriod)
+                // MaxDepositPeriod is not reached
+                
+                throw
+                
+              else
+                //  MaxDepositPeriod is reached 
+                
+                set deposit of sender in depositorsList to 0
+                sender.AtomBalance += deposit
+                
+            else
+              // Vote started
+              
+              retrieve initTotalVotingPower from txClaimDeposit.ProposalID
+              retrieve initProcedureNumber from txClaimDeposit.ProposalID    
+              retrieve initProcedure from initProcedureNumber // get procedure that was active when vote opened
+              
+              if  (proposal.Category AND proposal.Votes['Yes']/initTotalVotingPower >= 2/3) OR
+                  ((CurrentBlock > proposal.VotingStartBlock + initProcedure.VotingPeriod) AND (proposal.Votes['NoWithVeto']/(proposal.Votes['Yes']+proposal.Votes['No']+proposal.Votes['NoWithVeto']) < 1/3) AND           (proposal.Votes['Yes']/(proposal.Votes['Yes']+proposal.Votes['No']+proposal.Votes['NoWithVeto']) > 1/2)) then
+                
+                // Proposal was accepted either because
+                // Proposal was urgent and special condition was met
+                // Voting period ended and vote satisfies threshold
+                
+                set deposit of sender in depositorsList to 0
+                sender.AtomBalance += deposit
+                
+              else
+                throw
+```
+
+
+### Vote
+
+Once `ActiveProcedure.MinDeposit` is reached, voting period starts. From there, bonded Atom holders are able to send `TxVote` transactions to cast their vote on the proposal.
+
+```Go
+  type TxVote struct {
+    ProposalID           uint            //  proposalID of the proposal
+    Option               string          //  option from OptionSet chosen by the voter
+    ValidatorPubKey      crypto.PubKey   //  PubKey of the validator voter wants to tie its vote to
```

Comment:
We should draw more distintion here so folks know it's not the block signing key... I'd like ValGovernPubKey or something like that... get's confusing with `TxVote` as well... I think should maybe be called `TxGovVote`

Reply:
Actually it is the block signing key. I did that under the assumption that `GovernancePubKey` would be findable using `MainPubKey` but not the opposite. If this is the case, we need the main pub key in order to know how many Atoms are bonded to it. If this is not the case, then I can change the type and the associated pseudocode.

Ok for `TxGovVote` and other type names.